File: src/test/java/io/binac/leetcode/RestoreIPAddressesTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class RestoreIPAddressesTest {
    private final RestoreIPAddresses.Solution1 solution1 = new RestoreIPAddresses.Solution1();

    private final RestoreIPAddresses.Solution2 solution2 = new RestoreIPAddresses.Solution2();

    @Test
    void test1() {
        List<String> expected = Arrays.asList("255.255.11.135", "255.255.111.35");
        assertEquals(expected, solution1.restoreIpAddresses("25525511135"));
    }

    @Test
    void test2() {
        List<String> expected = Arrays.asList("255.255.11.135", "255.255.111.35");
        assertEquals(expected, solution2.restoreIpAddresses("25525511135"));
    }
}

File: cpp/unittests/RestoreIPAddressesTest.cpp
#include "RestoreIPAddresses.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(RestoreIPAddresses, Solution93_1) {
  const std::string S = "25525511135";
  const std::vector<std::string> Expected{
      "255.255.11.135", "255.255.111.35"
  };
  EXPECT_EQ(Expected, Solution93_1().restoreIpAddresses(S));
}